### DISCUSSION TITLE
How can I add withOptions using the Saloon package in Laravel?

### DISCUSSION THREAD

**Discussion by dhruva81** [saloonphp/saloon#211] [Q&A]
How can I add options using the Saloon package in Laravel? 

Previously, I was using Http::withOptions(['verify' => false]) to disable SSL verification when making requests.

But I am not able to find a way in Saloon package.

Kindly guide.

Thanks

**Comment by Sammyjo20** [comment]
Hey @dhruva81! Thanks for the question :) 

You may specify config options (Including Guzzle config options like `verify`) by using Saloon's `defaultConfig()` method on your connector / request or by using the `config()` method on the connector/request. Have a read here: https://docs.saloon.dev/the-basics/http-client-config

**Reply by dhruva81** [reply]
Thank you for your response.

Unfortunately, when the verify option is added to the request, the HTTP method changes to a GET request instead of a POST request. 

However, the method is explicitly defined as a POST request with the following code: `protected Method $method = Method::POST`.



Here is the current code:
```
class AttendanceReportRequest extends SoloRequest implements HasBody
{
    use HasJsonBody;

    public function __construct(
        protected array $params = []
    ) {
    }

    protected Method $method = Method::POST;

    public function resolveEndpoint(): string
    {
        return config('base_url').'/api/attendance';
    }

    protected function defaultHeaders(): array
    {
        return [
            'Content-Type' => 'application/json',
            'Accept' => 'application/json',
        ];
    }

    protected function defaultBody(): array
    {
        $userIds = User::all()->pluck('id')->toArray();

        return [
            'ext_ref_user_ids' => $userIds,
            ...$this->params,
        ];
    }

    public function createDtoFromResponse(Response $response): mixed
    {
        return AttendanceReportData::fromResponse($response);
    }
}

```

```
$request = AttendanceReportRequest::make($data);
$request->config()->add('verify', false);
$response = $request->send();
$records = $response->dto()->formattedData;
```
The resulting error message is:

```
"message" => "The GET method is not supported for route api/attendance. Supported methods: POST."
"exception" => "Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException"
```

**Reply by Sammyjo20** [reply]
Hey @dhruva81 apologies for the delay.

I will take a look at this an a potential issue with Saloon and I can see if I can replicate it 👍

**Comment by binaryfire** [comment]
Hi @dhruva81. Did you manage to resolve this?
